src/main/java/demo/ClusterCalphas.java

```diff
@@ -26,20 +29,24 @@
 	 */
 	public static void main(String[] args) throws IOException {
 		// Get the underlying C-alpha chains
-		List<Tuple2<String, Segment>> calphaChains = SparkUtils.getCalphaChains(new String[] {"1AQ1", "4CUP"}).filterMinLength(10).getSegmentRDD().collect();
-		int numChains = calphaChains.size();
+//		List<Tuple2<String, Segment>> calphaChains = SparkUtils.getCalphaChains(new String[] {"4CUP","4CUQ","4CUT","4CUR","4CUS","1STP"}).filterMinLength(10).getSegmentRDD().collect();
+		List<Tuple2<String, Segment>> listChains = new StructureDataRDD("/Users/anthony/full").sample(0.01).getCalpha().getSegmentRDD().collect();
+		int numChains = listChains.size();
+		Broadcast<List<Tuple2<String, Segment>>> calphaChains = SparkUtils.getSparkContext().broadcast(listChains);
+		System.out.println("Analysisng " + numChains + " chains");
 		// Now apply a function to those chains 
-		JavaRDD<Tuple3<Long, Long, Double>> similarities = SparkUtils.getComparisonMatrix(numChains).map(new TmScore(calphaChains));
+		JavaPairRDD<Integer, Integer> comparisons = SparkUtils.getComparisonMatrix(numChains);
+		System.out.println("Performing "+comparisons.count()+" comparisons");
+		JavaRDD<Tuple3<Long, Long, Double>> similarities = comparisons.map(new TmScore(calphaChains));
+		// Perform the clusterings
 		PowerIterationClustering pic = new PowerIterationClustering()
```

Comment:
and while we are at it, also v. interesting:

https://github.com/derrickburns/generalized-kmeans-clustering